dali/test/python/test_fw_iterators.py

```diff
@@ -149,6 +150,43 @@ def test_mxnet_iterator_last_batch_no_pad_last_batch():
     assert len(img_ids_list_set) == data_size
     assert len(set(mirrored_data)) != 1
 
+def test_mxnet_iterator_empty_array():
+    from nvidia.dali.plugin.mxnet import DALIGenericIterator as MXNetIterator
+
+    batch_size = 4
+    size = 5
+
+    def get_data():
+        images = [np.empty((1, 224, 224, 3), dtype = np.float)] * batch_size
+        bboxes = [np.empty((1, 0, 4), dtype = np.uint8)] * batch_size
+        labels = [np.array((1, 0, 1), dtype = np.longlong)] * batch_size
+        return (images, bboxes, labels)
+
+    pipe = Pipeline(batch_size=batch_size, num_threads=3, device_id=0)
+    images, bboxes, labels = fn.external_source(source = get_data, num_outputs = 3)
+    pipe.set_outputs(images, bboxes, labels)
+    pipe.build()
+
+    iterator = MXNetIterator(
+    pipe,
+    output_map=[
+        ('data',  MXNetIterator.DATA_TAG),
+        ('bboxes',  MXNetIterator.LABEL_TAG),
+        ('label',  MXNetIterator.LABEL_TAG)],
+    size=size,
+    dynamic_shape=True)
+
+    for i, batch in enumerate(iterator):
+        bboxes_shape = batch[0].label[0].asnumpy().shape
+
+        assert batch[0].data[0].asnumpy().dtype == np.float64
+        assert batch[0].label[0].asnumpy().dtype == np.uint8
+        assert batch[0].label[1].asnumpy().dtype == np.int64
+
+        assert bboxes_shape[0] == batch_size
+        assert bboxes_shape[1] == 0
+        assert bboxes_shape[2] == 4
```

Comment:
Can you also check the shape of other outputs?

Reply:
Done